packages/compass-aggregations/src/modules/aggregation.spec.ts

```diff
@@ -0,0 +1,288 @@
+import type { AnyAction, Store } from 'redux';
+import { createStore, applyMiddleware } from 'redux';
+import thunk from 'redux-thunk';
+import { expect } from 'chai';
+import { spy } from 'sinon';
+import type { Document } from 'mongodb';
+
+
+import reducer, { runAggregation, fetchNextPage, fetchPrevPage, cancelAggregation } from './aggregation';
+import type { State as AggregateState } from './aggregation';
+import type { RootState } from '.';
+import rootReducer, { INITIAL_STATE } from '../modules';
+import configureStore from '../stores/store';
+import { DATA_SERVICE_CONNECTED } from './data-service';
+
+const waitFor = (delay: number): Promise<void> => {
+  return new Promise(resolve => setTimeout(resolve, delay));
+};
+
+class AggregationCursorMock {
+  constructor(public options: {
+    hasNext: boolean;
+    documents: Document[];
+    delayInResponse?: number;
+    skip?: number;
+    limit?: number;
+  }) {
+  }
+  skip(count: number) {
+    this.options.skip = count;
+    return this;
+  }
+  limit(count: number) {
+    this.options.limit = count;
+    return this;
+  }
+  async hasNext() {
+    await waitFor(this.options.delayInResponse ?? 0);
+    return Promise.resolve(this.options.hasNext);
+  }
+  async toArray() {
+    await waitFor(this.options.delayInResponse ?? 0);
```

Comment:
I think these timers keep the test running – would it make sense to use `.unref()`ed timers for these? Can we avoid using timers altogether (it seems like what we really need here is just a Promise that does not resolve in time, which can also be a Promise that never fulfills or rejects)?